lib/active_merchant/billing/gateways/shift4.rb

```diff
@@ -153,7 +153,7 @@ def setup_access_token
         add_datetime(post, options)
 
         response = commit('accesstoken', post, request_headers('accesstoken', options))
-        raise ArgumentError, response.params.fetch('result', [{}]).first.dig('error', 'longText') unless response.success?
+        raise OAuthResponseError.new(response, response.params.fetch('result', [{}]).first.dig('error', 'longText')) unless response.success?
```

Comment:
Are there other failures possible? I'm concerned that this change would register all non-success statuses as an OAuth error when that might not apply 🤔 I would have had the same concern with the previous iteration.

Reply:
Good question! I see three possible paths in `setup_access_token`. 

1. Success
2. Failure due to invalid credentials
3. Failure due to the gateway being down

🤔 Right now both failures get reported as an OAuthResponseError but maybe I should explore trying to differentiate between the two? Can you think of any other paths that I may be missing?

Reply:
There could be a malformed request -- that would be caused by us, but hopefully caught in testing before merging. So I think starting out with differentiating between those two would be already accounting for the most probable errors 👍🏾

Reply:
Oh good, as long as the other "unacceptable" errors are handled elsewhere, I'm fine with what you have here 👍🏾